contracts/src/tests/test_multisig.cairo

```diff
@@ -697,23 +666,19 @@ fn test_recursive_set_signers_and_threshold() {
         );
 
     // Signer 1 confirms the transaction
-    set_caller_address(s1);
     set_contract_address(s1);
     multisig.confirm_transaction(0);
 
     // Signer 2 confirms the transaction
-    set_caller_address(s2);
```

Comment:
Hmm was this API removed?

Reply:
nah, it wasn't removed. we weren't using it the correct way. set_contract_address should only be used if the test contract is deployed in the testing suite. Otherwise we should use set_caller_address.

I noticed when testing that there is some undefined behavior if you do both